src/hooks/UrqlUseSubscriptionWithHandler.re

```diff
@@ -0,0 +1,72 @@
+[@bs.deriving abstract]
+type useSubscriptionWithHandlerArgs = {
+  query: string,
+  [@bs.optional]
+  variables: Js.Json.t,
+};
+
+[@bs.deriving abstract]
+type useSubscriptionWithHandlerResponseJs('acc) = {
+  fetching: bool,
+  [@bs.optional]
+  data: 'acc,
```

Comment:
Possible issue with type safety

Reply:
Can you explain a little more? I checked the type inference on the two hooks and they _appeared_ to be working as expected. `useSubscription` will infer the type of `data` as the result of calling `parse` on the `Js.Json.t` returned by `urql` (written as `'response`). `useSubscriptionWithHandler` will infer the type of the accumulator `'acc` you use in your `handler` function. We also already parse each `Js.Json.t` subscription coming back from the API before handing it off to `handler`, so the inference should work as:

1. Infer the type of `'response` from calling `parse`.
2. Infer the type of `'acc` via the `handler` function (i.e. does a user supply an array, a `Js.Dict`, etc. for accumulating new subscriptions).

This piece is somewhat complex tho and I could definitely be missing something. Thannks as always for a thorough review ❤️

Reply:
You're completely right! I hadn't thought that `useSubscriptionWithHandlerJs` is safely wrapped through `useSubscriptionHandler`

Maybe it would be good to comment this for future contributors to take notice? Because it can still be used in a non-type safe manner, but not by the library users, only inside this module. I believe it would be good to refactor a little bit so that the unsafe surface area is minimized